staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -302,29 +302,49 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		// with non-equal specs (especially non-equal schemas).
 		var crds []*apiextensionsv1.CustomResourceDefinition
 		crds, err = r.crdLister.List(labels.Everything())
-		if err == nil {
-			if len(crds) == 0 {
-				err = errors.NewNotFound(schema.GroupResource{Group: apiextensionsv1.SchemeGroupVersion.Group, Resource: "customresourcedefinitions"}, "")
+		if err != nil {
```

Comment:
Error common management and return (with a special treatment for `NotFound` errors) is done later, at line [353](https://github.com/kcp-dev/kubernetes/pull/20/files#diff-01e7a2059fe62e95ec32e2589c01b7218100de327647c641c334d38f2c5aa06eR353).
So isn't the fix simpler here: just set the `err` to `NotFound` in line [340](https://github.com/kcp-dev/kubernetes/pull/20/files#diff-01e7a2059fe62e95ec32e2589c01b7218100de327647c641c334d38f2c5aa06eR340) when no valid CRD has been found ?

Reply:
cf my answer to your comment on my initial PR: https://github.com/kcp-dev/kubernetes/pull/15#discussion_r743009482